sources/api/schnauzer/src/helpers.rs

```diff
@@ -1486,30 +1596,14 @@ fn oci_spec_capabilities(value: &Value) -> Result<String, RenderError> {
 /// This helper function generates the resource limits section of
 /// the OCI runtime spec from the provided `value` parameter, which is
 /// the settings data from the datastore (`settings.oci-defaults.resource-limits`).
-fn oci_spec_resource_limits(value: &Value) -> Result<String, RenderError> {
-    let oci_default_rlimits: HashMap<OciDefaultsResourceLimitType, OciDefaultsResourceLimit> =
-        serde_json::from_value(value.clone())?;
-
-    let result_lines = oci_default_rlimits
-        .iter()
-        .map(|(rlimit_type, values)| {
-            format!(
-                r#"{{ "type": "{}", "hard": {}, "soft": {} }}"#,
-                rlimit_type.to_linux_string(),
-                values.get_hard_limit(),
-                values.get_soft_limit(),
-            )
-        })
-        .collect::<Vec<String>>()
-        .join(",\n");
-
-    debug!("resource_limits result_lines: \n{}", result_lines);
-    Ok(result_lines)
+fn oci_spec_resource_limits(
+    value: &Value,
+) -> Result<HashMap<OciDefaultsResourceLimitType, OciDefaultsResourceLimit>, RenderError> {
+    Ok(serde_json::from_value(value.clone())?)
 }
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 // helpers to the helpers
-
 /// Gets the key name (path) of the parameter at `index`. Returns an error if the param cannot be extracted.
```

Comment:
nit: try not to make unnecessary style changes.

```suggestion

/// Gets the key name (path) of the parameter at `index`. Returns an error if the param cannot be extracted.
```